addons/spreadsheet_dashboard/static/src/bundle/dashboard_action/dashboard_action.js

```diff
@@ -64,6 +65,25 @@ export class SpreadsheetDashboardAction extends Component {
                 return [dashboard && dashboard.model, dashboard && dashboard.status];
             }
         );
+        useEffect(() => {
+            const printLog = (event) => {
+                const dashboard = this.state.activeDashboard;
+                if (!dashboard || dashboard.status !== Status.Loaded) {
+                    return;
+                }
+                const datasources = dashboard.model.getters
+                    .getLoadedDatasources()
+                    .map((ds) => ds.source);
```

Comment:
It looks like all uses of `getLoadedDatasources` are followed by `.map((ds) => ds.source)` -> it could return what's needed instead